pkg/git/git.go

```diff
@@ -97,6 +120,21 @@ func (git Client) DeleteLocalRepo() error {
 	return fmt.Errorf("no cloned directory to delete")
 }
 
+// RepoName returns the name of the repository given its URL
+func RepoName(repoURL string) (string, error) {
+	// FIXME: urls like git@github.com:weaveworks/eksctl produce the error "first path segment in URL cannot contain colon"
+	u, err := url.Parse(strings.ReplaceAll(repoURL, ":", "/"))
```

Comment:
Duplicate: `strings.ReplaceAll(repoURL, ":", "/")`
It may be worth having a `SanitizeGitURL` function factorising this. And ideally some unit tests to cover it and add more context on what we're trying to achieve there, and why.

Reply:
fixed in new helper functions in the git package that call a library to parse the git urls